mmdet/models/detectors/maskformer.py

```diff
@@ -0,0 +1,103 @@
+import warnings
+
+from ..builder import DETECTORS, build_backbone, build_head, build_neck
+from .single_stage import SingleStageDetector
+
+
+@DETECTORS.register_module()
+class MaskFormer(SingleStageDetector):
+    r"""Implementation of `Per-Pixel Classification is
+    NOT All You Need for Semantic Segmentation
+    <https://arxiv.org/pdf/2107.06278>`"""
+
+    def __init__(self,
+                 backbone,
+                 neck=None,
+                 semantic_head=None,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None,
+                 init_cfg=None):
+        super(SingleStageDetector, self).__init__(init_cfg=init_cfg)
+        if pretrained:
+            warnings.warn('DeprecationWarning: pretrained is deprecated, '
+                          'please use "init_cfg" instead')
+            backbone.pretrained = pretrained
+        self.backbone = build_backbone(backbone)
+        if neck is not None:
+            self.neck = build_neck(neck)
+        semantic_head.update(train_cfg=train_cfg)
+        semantic_head.update(test_cfg=test_cfg)
+        self.semantic_head = build_head(semantic_head)
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+
+    def extract_feat(self, img):
```

Comment:
Line 35-40 can be deleted as it is the same as the original extract_feat in single stage detector.